models/Element/Recyclebin/Item.php

```diff
@@ -349,15 +318,23 @@ function ($currentValue) {
                 }
             ),
             new class($this->element) extends TypeMatcher {
+
+                protected mixed $element;
+                public function __construct($element)
+                {
+                    $this->element = $element;
+                    parent::__construct($element);
+                }
+
                 /**
                  * @param mixed $element
                  *
                  * @return bool
                  */
-                public function matches($element)
+                public function matches($element): bool
                 {
                     //compress only elements with full_dump_state = false
-                    return $element instanceof Element\ElementInterface && $element instanceof Element\ElementDumpStateInterface && !($element->isInDumpState());
+                    return $this->element instanceof Element\ElementInterface && $this->element instanceof Element\ElementDumpStateInterface && !($this->element->isInDumpState());
```

Comment:
why changed from `$element` to `$this->element`

Reply:
I had another look at the code and changed it a bit, we can cast `$element` directly to string (because __toString() method is part of the interface), then there is no need for the other stuff.